### PR TITLE
build: update groovy wrapper

### PR DESCRIPTION
**Pull request by jdrueckert**
### What this PR does

Consume https://github.com/MovingBlocks/groovy-wrapper/releases/tag/v2.2.0 as documented in https://github.com/MovingBlocks/groovy-wrapper#to-use

### ToDo before merge

- [X] Figure out what I'm doing wrong and fixing it :roll_eyes: 

Attempt 1:
What I did: Follow the "download, extract and place" approach (plus `chmod +x`).
What's going wrong: Locally running `./groovyw usage` currently fails with `/usr/bin/env: ‘sh\r’: No such file or directory`.

Attempt 2:
What I did: Follow the "just grab and add" (copy + paste) approach.
What's going wrong: Locally running `./groovyw usage` currently fails with
```
Caught: java.io.FileNotFoundException: /path/to/ts/workspace/usage (/path/to/ts/workspace/usage)
java.io.FileNotFoundException: /path/to/ts/workspace/usage (/path/to/ts/workspace/usage)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.gradle.wrapper.GroovyBootstrapMainStarter.start(GroovyBootstrapMainStarter.java:36)
	at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:103)
	at org.gradle.wrapper.GroovyWrapperMain.main(GroovyWrapperMain.java:60)
```

### CODE REVIEW

**Comment by BenjaminAmos** [comment]
> Locally running ./groovyw usage currently fails with /usr/bin/env: ‘sh\r’: No such file or directory.

On Windows I got the following after cloning https://github.com/MovingBlocks/Terasology/pull/5141/commits/51187f7c3d1d266d7602e0d9b7b0a68f5e7cfc05:
```
Caught: java.io.FileNotFoundException: <workspace path>\Terasology\module (<workspace path>\Terasology\module)
java.io.FileNotFoundException: <workspace path>\Terasology\module (<workspace path>\Terasology\module)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at org.gradle.wrapper.GroovyBootstrapMainStarter.start(GroovyBootstrapMainStarter.java:36)
        at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:103)
        at org.gradle.wrapper.GroovyWrapperMain.main(GroovyWrapperMain.java:60)
```

**Comment by jdrueckert** [comment]
> > Locally running ./groovyw usage currently fails with /usr/bin/env: ‘sh\r’: No such file or directory.
> 
> On Windows I got the following after cloning [51187f7](https://github.com/MovingBlocks/Terasology/commit/51187f7c3d1d266d7602e0d9b7b0a68f5e7cfc05):
> 
> ```
> Caught: java.io.FileNotFoundException: <workspace path>\Terasology\module (<workspace path>\Terasology\module)
> java.io.FileNotFoundException: <workspace path>\Terasology\module (<workspace path>\Terasology\module)
>         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>         at org.gradle.wrapper.GroovyBootstrapMainStarter.start(GroovyBootstrapMainStarter.java:36)
>         at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:103)
>         at org.gradle.wrapper.GroovyWrapperMain.main(GroovyWrapperMain.java:60)
> ```

Yup, that's the same I get now too after copy/pasting the files rather than taking them from the extracted release zip... not entirely sure whether I'm doing something wrong or the documentation in the repo readme is out of date or whatever..

Interesting, though, that you got this already with my first attempt.

File: groovyw.bat [hunk 1]
```diff
@@ -112,7 +86,7 @@ set CLASSPATH=%APP_HOME%\gradle\wrapper\gradle-wrapper.jar;%APP_HOME%\gradle\wra
 
 
 @rem Execute Groovy via the Gradle Wrapper
-"%JAVA_EXE%" %DEFAULT_JVM_OPTS% %JAVA_OPTS% %GRADLE_OPTS% "-Dorg.gradle.appname=%APP_BASE_NAME%" -classpath "%CLASSPATH%" org.gradle.wrapper.GroovyWrapperMain config/groovy/util.groovy %CMD_LINE_ARGS%
+"%JAVA_EXE%" %DEFAULT_JVM_OPTS% %JAVA_OPTS% %GRADLE_OPTS% "-Dorg.gradle.appname=%APP_BASE_NAME%" -classpath "%CLASSPATH%" org.gradle.wrapper.GroovyWrapperMain %CMD_LINE_ARGS%
```
Comment (BenjaminAmos) [inline groovyw.bat]: This is the issue. This line should not have been changed. It works for me when I revert that change.

File: groovyw [hunk 2]
```diff
@@ -180,6 +180,6 @@ save () {
 APP_ARGS=`save "$@"`
 
 # Collect all arguments for the java command, following the shell quoting and substitution rules
-eval set -- $DEFAULT_JVM_OPTS $JAVA_OPTS $GRADLE_OPTS "\"-Dorg.gradle.appname=$APP_BASE_NAME\"" -classpath "\"$CLASSPATH\"" org.gradle.wrapper.GroovyWrapperMain config/groovy/util.groovy "$APP_ARGS"
+eval set -- $DEFAULT_JVM_OPTS $JAVA_OPTS $GRADLE_OPTS "\"-Dorg.gradle.appname=$APP_BASE_NAME\"" -classpath "\"$CLASSPATH\"" org.gradle.wrapper.GroovyWrapperMain "$APP_ARGS"
```
Comment (BenjaminAmos) [inline groovyw]: This is the issue. This line should not have been changed. It works for me when I revert that change.

**Review by BenjaminAmos — DISMISSED** [review]

File: groovyw.bat (hunk 1 above)
Comment (jdrueckert) [reply groovyw.bat]: but then that must've already gone wrong in https://github.com/MovingBlocks/groovy-wrapper, right?

Comment (jdrueckert) [reply groovyw.bat]: or rather it might be specific to how we use the wrapper in the engine...

Comment (BenjaminAmos) [reply groovyw.bat]: I believe that it was behaving as intended. When you run `groovyw Test` to test the wrapper in its own repository, it is running `Test.groovy`. The first argument to the wrapper is the groovy script to run, I think.

**Comment by jdrueckert** [comment]
@BenjaminAmos thanks for the hint, works now.

**Review by Cervator — APPROVED** [review]
Can validate that this works now. I submitted https://github.com/MovingBlocks/groovy-wrapper/commit/e9ee6672607c25da73757b2349048450ba52c4e2 to better document the customization of the Groovy Wrapper for a given project. 

Still left out is another curiosity that might be a thing for another day: when the wrapper's base Gradle changes and the `gradlew` + `gradlew.bat` files get modified _in theory_ those changes should be replicated in the `groovyw` + `groovyw.bat` files - but it seems unlikely that the thinner scope of simply running Groovy scripts will be affected every time, and this time it clearly isn't since, well, it works! I guess it sometimes is more about edge cases. Having upgraded the wrapper so rarely it just isn't something I've had much of a chance to think on, and really the number one thing is "Does it work" so ... it does!

**Pull request merged**